pkg/admin/migration/types.go

```diff
@@ -0,0 +1,70 @@
+package migration
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type MigratonStatus string
+
+const (
+	MigrationSuccess MigratonStatus = "success"
+	MigrationFailure MigratonStatus = "failure"
+)
+
+// Container for individual migration attempts
+type MigrationResult struct {
+	Error           error                       `json:"error,omitempty"`
+	ResourceVersion schema.GroupVersionResource `json:"resourceVersion"`
+	Timestamp       time.Time                   `json:"timestamp"`
+	ObjectMeta      *metav1.ObjectMeta          `json:"object,omitempty"`
```

Comment:
We could think about using [NamespacedName](https://github.com/kubernetes/apimachinery/blob/master/pkg/types/namespacedname.go#L27) if this is what we need from this

Reply:
Oh right, perfect, thanks!